lib/internal/streams/writable.js

```diff
@@ -176,6 +190,85 @@ ObjectDefineProperties(WritableState.prototype, {
 
   allBuffers: makeBitMapDescriptor(kAllBuffers),
   allNoop: makeBitMapDescriptor(kAllNoop),
+
+  // Indicates whether the stream has errored. When true all write() calls
+  // should return false. This is needed since when autoDestroy
+  // is disabled we need a way to tell whether the stream has failed.
+  // This is/should be a cold path.
+  errored: {
+    __proto__: null,
+    enumerable: false,
+    get() { return (this.state & kErrored) !== 0 ? this[kErroredValue] : null; },
+    set(value) {
+      if (value) {
+        this[kErroredValue] = value;
+        this.state |= kErrored;
+      } else {
+        this.state &= ~kErrored;
+      }
+    },
+  },
+
+  writable: {
+    __proto__: null,
+    enumerable: false,
+    get() { return (this.state & kHasWritable) !== 0 ? (this.state & kWritable) !== 0 : undefined; },
+    set(value) {
+      if (value == null) {
+        this.state &= ~(kHasWritable | kWritable);
+      } else if (value) {
+        this.state |= (kHasWritable | kWritable);
+      } else {
+        this.state |= kHasWritable;
+        this.state &= ~kWritable;
```

Comment:
if it has performance improvement:
```suggestion
this.state = (this.state | kHasWritable) & ~kWritable;
```

Reply:
I doubt it but always better to check the IR